cirq-web/cirq_ts/src/circuit/grid_circuit.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Group} from 'three';
+import {GridQubit} from './components/grid_qubit';
+import {TwoQubitGate, SingleQubitGate, GridCoord} from './components/types';
+
+/**
+ * Class that gathers serialized circuit information
+ * from a Python cirq.Circuit and reconstructs it to be
+ * displayed using three.js
+ */
+export class GridCircuit extends Group {
+  readonly moments: number;
+  private circuit: Map<string, GridQubit>;
+
+  /**
+   * Class constructor
+   * @param moments The number of moments of the circuit. This
+   * determines the length of all the qubit lines in the diagram.
+   * @param qubits A list of GridCoord objects representing the locations of the
+   * qubits in the circuit.
+   */
+  constructor(moments: number, qubits: GridCoord[]) {
+    super();
+    this.moments = moments;
+    this.circuit = new Map();
+
+    for (const coord of qubits) {
+      this.addQubit(coord.row, coord.col);
+    }
+  }
+
+  /**
+   * Adds Gate objects to class circuit map and generates three.js
+   * objects, adding them to the group.
+   * @param list A list of Gate objects to be added to the circuit.
+   */
+  displayGatesFromList(list: (SingleQubitGate | TwoQubitGate)[]) {
```

Comment:
The `FromList` is clear from the signature, so I think that's fine to omit from the method name.  Since you're "add"ing the gates and not necessarily "displaying" them consider something like `addGates`?